modelplace_api/visualization.py

```diff
@@ -490,80 +521,50 @@ def draw_classification_result(
 def draw_age_gender_recognition_result(
     image: Union[Image, np.ndarray], detections: List[AgeGenderLabel],
 ) -> List[np.ndarray]:
-    image = np.ascontiguousarray(image)
-    image_with_bbox = image.copy()
-    images = []
-    for detection in detections:
-        image_with_bbox = cv2.rectangle(
-            image_with_bbox,
-            (int(detection.bbox.x1), int(detection.bbox.y1)),
-            (int(detection.bbox.x2), int(detection.bbox.y2)),
-            RGB_COLORS[17],
-            thickness=8,
-        )
+    image_with_boxes = np.ascontiguousarray(image)
+    source_image = image_with_boxes.copy()
+    images = list()
+    images.append(source_image)
     for detection in detections:
-        image = image_with_bbox.copy()
-        image = cv2.rectangle(
-            image,
+        image_with_boxes = cv2.rectangle(
+            image_with_boxes,
             (int(detection.bbox.x1), int(detection.bbox.y1)),
             (int(detection.bbox.x2), int(detection.bbox.y2)),
             RGB_COLORS[196],
             thickness=8,
         )
-        classification_labels = {
-            label.class_name.capitalize(): round(float(label.score), 2)
-            for label in detection.gender
-        }
-        classification_labels["Age"] = detection.age
-        image = draw_classification_legend(
-            image,
-            classification_labels,
-            font_scale=0.75,
-            font_thickness=2,
-            height_offset=20,
-            cell_height=50,
+        bg_color = RGB_COLORS[196]
+        label_score, label_class_name = detection.gender[0]
+        age = detection.age
+        text = f"{label_class_name[1]}, Age: {age}"
+        image_with_boxes = draw_text_label(
+            image_with_boxes, detection, text, bg_color=bg_color,
         )
-        images.append(image)
+        images.append(image_with_boxes)
     return images
 
 
 def draw_emotion_recognition_result(
     image: Union[Image, np.ndarray], detections: List[EmotionLabel],
 ) -> List[np.ndarray]:
-    image = np.ascontiguousarray(image)
-    image_with_bbox = image.copy()
-    images = []
-    for detection in detections:
-        image_with_bbox = cv2.rectangle(
-            image_with_bbox,
-            (int(detection.bbox.x1), int(detection.bbox.y1)),
-            (int(detection.bbox.x2), int(detection.bbox.y2)),
-            RGB_COLORS[17],
-            thickness=8,
-        )
+    image_with_boxes = np.ascontiguousarray(image)
+    source_image = image_with_boxes.copy()
+    images = list()
+    images.append(source_image)
     for detection in detections:
-        image = image_with_bbox.copy()
-        image = cv2.rectangle(
-            image,
+        image_with_boxes = cv2.rectangle(
+            image_with_boxes,
             (int(detection.bbox.x1), int(detection.bbox.y1)),
             (int(detection.bbox.x2), int(detection.bbox.y2)),
             RGB_COLORS[196],
             thickness=8,
```

Comment:
Maybe better to specify as a default argument?

@PVSemk what do you think?

Reply:
Resonable, as we did it earlier: [link](https://github.com/opencv-ai/modelplace-api/pull/9/files/fd65183bc05a04afed2764aaeb8fdbe804490ddd#diff-63fde0013d268f8f6319418745f702a12000cff7c559b8c3347b6eed32128077R75)